ablySpec/RealtimeClient.connection.swift

```diff
@@ -648,6 +648,78 @@ class RealtimeClientConnection: QuickSpec {
                         expect(nacks[0].msgSerial).to(equal(6))
                         expect(nacks[0].count).to(equal(1))
                     }
+                }
+
+                // RTN7c
+                context("should deliver as failed the remaining pending messages if") {
+
+                    it("connection is closed") {
+                        client.connect()
+                        defer {
+                            client.dispose()
+                            client.close()
+                        }
+
+                        let channel = client.channel("channel")
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.subscribeToStateChanges { state, status in
+                                if state == .Attached {
+                                    channel.publish("message", cb: { status in
+                                        expect(status.state).to(equal(ARTState.Error))
+                                        expect(status.errorInfo!.code).to(equal(80003))
+                                        done()
+                                    })
+                                    client.close()
```

Comment:
> It isn't an async call. Even so, I will check that.

Exactly, because it is async unfortunately there are no guarantees about what has happened i.e. it could be in the queue or it could be sent and waiting an `N?ACK`